modules/navier_stokes/src/fvbcs/INSFVMassAdvectionOutflowBC.C

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVMassAdvectionOutflowBC.h"
+#include "INSFVVelocityVariable.h"
+#include "SubProblem.h"
+#include "MooseMesh.h"
+
+registerADMooseObject("NavierStokesApp", INSFVMassAdvectionOutflowBC);
+
+InputParameters
+INSFVMassAdvectionOutflowBC::validParams()
+{
+  InputParameters params = FVFluxBC::validParams();
+  params += INSFVFullyDevelopedFlowBC::validParams();
+  params.addClassDescription("Outflow boundary condition for advecting mass.");
+  params.addRequiredParam<Real>("rho", "The value for the density");
+  params.declareControllable("rho");
+  params.addRequiredCoupledVar("u", "The velocity in the x direction.");
+  params.addCoupledVar("v", "The velocity in the y direction.");
+  params.addCoupledVar("w", "The velocity in the z direction.");
+  return params;
+}
+
+INSFVMassAdvectionOutflowBC::INSFVMassAdvectionOutflowBC(const InputParameters & params)
+  : FVFluxBC(params),
+    INSFVFullyDevelopedFlowBC(params),
+    _rho(getParam<Real>("rho")),
+    _u_var(dynamic_cast<const INSFVVelocityVariable *>(
+        &_subproblem.getVariable(_tid, params.get<std::vector<VariableName>>("u").front()))),
+    _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(&_subproblem.getVariable(
```

Comment:
any reason you are preferring this over getFieldVar?

Reply:
No, not really. I suppose that I could save a THREAD_ID argument by calling `getFieldVar` :smile: I can make that change

Reply:
Oh and also save the extra params.get.*front fenagling